tds.go

```diff
@@ -644,6 +662,64 @@ func ReadLoginRequest(_r io.ReadWriteCloser) (*LoginRequest, error) {
 	}, nil
 }
 
+// ReadLoginResponse parses a TDS7 login response packet.
+func ReadLoginResponse(_r io.ReadWriteCloser) (l *LoginResponse, err error) {
+	defer func() {
```

Comment:
This `defer func()...` pattern also appears in ReadError below. Is it possible to consolidate by creating a helper function, and then defer on that helper, or can recover only be called directly in a defer?